### DISCUSSION TITLE
What's the difference between explain_message and edge_weight?

### DISCUSSION THREAD

**Discussion by ZechenGao** [pyg-team/pytorch_geometric#9506] [Q&A] [answered]
I'm new to pytorch_geometric explain module. I found def explain_message(in message_passing.py) multiplies self._edge_mask and the out after message propagating. And edge_weight seems like dealing with the same way(eg. def message in gcn_conv.py). What's the diffrence between them? Thanks!

**Comment by kano5266** [comment, marked as answer]
Hi there!
The `edge_mask` in `expain_message` doesn't change the calculation of the node degree $`\hat{d}_i = 1 + \sum_{j \in \mathcal{N}(i)} e_{j,i}`$ in GCNConv. However, the edge weight does affect it. 

Also, `edge_mask` can be used on all massage passing neural networks. While only a few MPNNs accept edge_weight as input.

Hope that helps!

**Reply by ZechenGao** [reply]
Thanks! That helps me a lot!
